app/Models/Transaction.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Str;
+
+class Transaction extends Model
+{
+    protected $guarded = [];
+
+    // Transaction Types
+    const DEBIT = 1;
+    const CREDIT = 2;
+
+    // Transaction Status
+    const SUCCESS = 1;
+    const FAILED = 2;
+    const REFUND = 3;
+    const AWAITING_PAYMENT = 4;
+    const PREPARING = 5;
+    const TO_SHIP = 6;
+    const TO_RECEIVE = 7;
+
+    public static function boot() {
+        parent::boot();
+
+        self::creating(function($user)
+        {
+            // TODO: Create a tracking number generator
+            $user->tracking_number = Str::uuid();
```

Comment:
I think uuid is already okay with tracking number? do you have any suggestions?

Reply:
We should have a human readable tracking number,
- can be dictated in case of followup